### PR TITLE
NAS-132298 / 25.04 / Add Delete and Edit actions for virtualization instances

### PR DESCRIPTION
**Pull request by AlexKarpov98**
Testing: 

See ticket.
I fixed initial list instances loading as well, it didn't work correctly.

Known issue: Auto-start is always TRUE (unchangeable) -> William is notified about it

Result:

https://github.com/user-attachments/assets/7a97378b-d6af-4750-8019-c900c0deee23

### CODE REVIEW

**Review by undsoft — CHANGES_REQUESTED** [review]
There is no need to extend instance-wizard for edit scenario. Creating and editing actions are sufficiently different for a new edit component to show in slide in.

**Comment by AlexKarpov98** [comment]
Edit form now shows up in Slide In.

Screenshot 2024-11-07 at 22 58 35

File: src/app/pages/virtualization/components/all-instances/instance-list/instance-list.component.ts [hunk 1]
```diff
@@ -89,14 +90,20 @@ export class InstanceListComponent implements AfterViewInit {
...
-  ) {}
-
-  ngAfterViewInit(): void {
-    this.activatedRoute.paramMap.pipe(
-      untilDestroyed(this),
-    ).subscribe((params) => {
-      this.selectForDetails(params.get('id'));
-    });
+  ) {
+    effect(() => {
```
Comment (undsoft) [inline src/app/pages/virtualization/components/all-instances/instance-list/instance-list.component.ts:94]: Let's move it into its own class field.

File: src/app/pages/virtualization/components/instance-wizard/select-image-dialog/select-image-dialog.component.ts [hunk 2]
```diff
@@ -75,7 +75,7 @@ export class SelectImageDialogComponent implements OnInit {
 
   constructor(
     private ws: WebSocketService,
-    private dialogRef: MatDialogRef<CreateInstanceFormComponent>,
+    private dialogRef: MatDialogRef<InstanceWizardComponent>,
```
Comment (undsoft) [inline src/app/pages/virtualization/components/instance-wizard/select-image-dialog/select-image-dialog.component.ts:78]: Actually incorrect reference here. DialogRef refers to the component that was open, so SelectImageDialog in this case.

File: src/app/interfaces/api/api-call-directory.interface.ts [hunk 3]
```diff
@@ -832,6 +832,7 @@ export interface ApiCallDirectory {
 
   // Virt
   'virt.instance.query': { params: QueryParams<VirtualizationInstance>; response: VirtualizationInstance[] };
+  'virt.instance.get_instance': { params: [instanceId: string]; response: VirtualizationInstance };
```
Comment (undsoft) [inline src/app/interfaces/api/api-call-directory.interface.ts:835]: Not used, can be removed.

File: src/app/pages/virtualization/components/instance-wizard/instance-wizard.component.ts [hunk 4]
```diff
@@ -92,19 +107,17 @@ export class CreateInstanceFormComponent {
...
       title: this.translate.instant('Saving Instance'),
     })
       .afterClosed()
       .pipe(untilDestroyed(this))
       .subscribe({
-        next: (newInstance) => {
+        next: ({ result }) => {
           this.snackbar.success(this.translate.instant('Instance saved'));
-          this.router.navigate(['/virtualization/view', newInstance.id]);
+          this.router.navigate(['/virtualization/view', result?.id]);
```
Comment (undsoft) [inline src/app/pages/virtualization/components/instance-wizard/instance-wizard.component.ts:120]: Hah, interesting bug.

**Review by undsoft — APPROVED** [review]
Save Instance button is slightly misaligned with the rest of the form, otherwise looks goo d – I was able to edit and delete a container.

**Pull request merged**